libs/shared/shared_common.c

```diff
@@ -1,8 +1,72 @@
 // Copyright (c) Microsoft Corporation
 // SPDX-License-Identifier: MIT
 
+#include "ebpf_program_types.h"
 #include "ebpf_shared_framework.h"
 
+static bool
+_ebpf_validate_helper_function_prototype(const ebpf_helper_function_prototype_t* helper_prototype)
+{
+    return (
+        (helper_prototype != NULL) && (helper_prototype->header.size >= EBPF_HELPER_FUNCTION_PROTOTYPE_MINIMUM_SIZE) &&
+        (helper_prototype->header.version <= EBPF_HELPER_FUNCTION_PROTOTYPE_VERSION_LATEST) &&
+        (helper_prototype->name != NULL));
+}
+
+bool
+ebpf_validate_helper_function_prototype_array(
+    _In_reads_(count) const ebpf_helper_function_prototype_t* helper_prototype, uint32_t count)
+{
+    if (count > 0) {
+        for (uint32_t i = 0; i < count; i++) {
+            if (!_ebpf_validate_helper_function_prototype(&helper_prototype[i])) {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+static bool
+_ebpf_validate_context_descriptor(_In_ const ebpf_context_descriptor_t* context_descriptor)
+{
+    return ((context_descriptor != NULL) && (context_descriptor->size >= sizeof(ebpf_context_descriptor_t)));
```

Comment:
Why no version check here?

Reply:
Context descriptor is used in `EbpfHelperPrototype`. If you think this should be versioned, I will create a separate issue to track this.